lib/matplotlib/widgets.py

```diff
@@ -2577,6 +2582,16 @@ def _contains(self, event):
         """Return True if event is within the patch."""
         return self._selection_artist.contains(event, radius=0)[0]
 
+    @staticmethod
+    def _snap(values, snap_values):
+        """Snap values to a given array values (snap_values)."""
+        indices = np.empty_like(values, dtype="uint")
+        # take into account machine precision
+        eps = np.min(np.abs(np.diff(snap_values))) * 1e-12
+        for i, v in enumerate(values):
+            indices[i] = np.abs(snap_values - v + np.sign(v) * eps).argmin()
+        return snap_values[indices]
```

Comment:
```suggestion
        indices = np.empty_like(values, dtype="uint")
        # take into account machine precision
        eps = np.min(np.abs(np.diff(snap_values))) * 1e-12
        return tuple(
            snap_values[np.abs(snap_values - v + np.sign(v) * eps).argmin()]
            for v in values)
```
is maybe a bit easier as you don't have to carry the indices.